src/api/api.jl

```diff
@@ -81,14 +81,25 @@ end
 
 const use_stdcall = startswith(basename(libmpi), "msmpi")
 
-macro mpicall(expr)
-    @assert expr isa Expr && expr.head == :call && expr.args[1] == :ccall
+macro mpicall(expr::Expr)
+    is_ccall = expr.head == :call && expr.args[1] == :ccall
+    is_at_ccall = expr.head == :macrocall && expr.args[1] == Symbol("@ccall")
+    if !is_ccall && !is_at_ccall
+        error("@mpicall can only be applied to `ccall`/`@ccall` expressions!")
+    end
 
     # On unix systems we call the global symbols to allow for LD_PRELOAD interception
     # It can be emulated in Windows (via Libdl.dllist), but this is not fast.
-    if Sys.isunix() && expr.args[2].head == :tuple &&
-            (VERSION ≥ v"1.5-" || expr.args[2].args[1] ≠ :(:MPI_Get_library_version))
```

Comment:
Am I correct to say that this line is always true since we support Julia v1.6+ now?